### ISSUE TITLE
Procedural Dome/Hemisphere mesh

### ISSUE THREAD

**Issue by LeCloutPanda** [Yellow-Dog-Man/Resonite-Issues#3165]
### Is your feature request related to a problem? Please describe.

Lack of Dome/Hemisphere meshes to make domes, caps, more detailed models has forced some to use external tools or in game material tricks to get it working

### Describe the solution you'd like

A new procedural mesh component that has variables listed below and just what ever other ones are needed
- Radius
- Hollow center(Bool)
- Wall thickness(to be used with the setting above)
- detail settings like the Sphere mesh
- etc

### Describe alternatives you've considered

Using a sphere mesh with a interest material and a ring mesh to fill in the gap that intersect material leaves.

### Additional Context

The alternative I considered above is being used by my self and I have no doubt others have figured the same but when in motion the effect falls apart while the material catches up.

### Requesters

[LeCloutPanda(Discord)](https://github.com/LeCloutPanda)

**Comment by coolymike** [comment]
Any progress on this?

**Comment by Frooxius** [comment]
No. It hasn't been picked up or assigned by anyone as you can see on the assignees. 

We'll share progress if we have progress.
